### PR TITLE
#1666: `overrideMappingTableOwnFilter` in `MappingConformanceRule` makes its old serialization incompatible

### LINKED ISSUE #1666: `overrideMappingTableOwnFilter` in `MappingConformanceRule` makes its old serialization incompatible
## Describe the bug
The new property `MappingConformanceRule.overrideMappingTableOwnFilter` in  defined as not optional. That makes it old serialization incompatible with the new definition - property value os considered `null`.

## To Reproduce
Steps to reproduce the behavior OR commands run:
1. Have a pre 2.18 serialziation of  `MappingConformanceRule`
2. Load it
3. Try to see it
4. See error (attached picture)
image

## Expected behaviour
No error

### PR DESCRIPTION
**Pull request by benedeki**
* `overrideMappingTableOwnFilter` made optional
* added UT testing the old serialization
* fixed a few compilation warnings
Fixes #1666 

Release notes:
`overrideMappingTableOwnFilter` was made optional in `MappingConformanceRule` de-/serialization so old serializations of conformance rules without the property are still readable.

### CODE REVIEW

File: data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala [hunk 1]
```diff
@@ -153,7 +157,8 @@ package object conformanceRule {
 
   object MappingConformanceRule {
     // attributeMappings property has key's with dot's that mongo doesn't accept; this symbol is used to replace the dots
-    val DOT_REPLACEMENT_SYMBOL: Char = '^'
+    final val DOT_REPLACEMENT_SYMBOL: Char = '^'
+    final val DEFAULT_OVERRIDE_MAPPING_TABLE_OWN_FILTER: Boolean = false
```
Comment (dk1844) [inline data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala:161]: I don't think this is the proper letter casing in Scala for constants. I believe `CapitalizedCamelCase` should be used

**Review by dk1844 — APPROVED** [review]
Otherwise LGTM.

File: data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala (hunk 1 above)
Comment (benedeki) [reply data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala:161]: I agree. But I adhered to the existing constant. On the other hand, why not to change both of them... 🤔

Comment (AdrianOlosutean) [reply data-model/src/main/scala/za/co/absa/enceladus/model/conformanceRule/package.scala:161]: I also had the same first impression: [here](https://docs.scala-lang.org/style/naming-conventions.html)

**Review by dk1844 — APPROVED** [review]

**Review by Zejnilovic — APPROVED** [review]
Code Reviewed

**Review by Zejnilovic — APPROVED** [review]
Built and Run. Menas no longer has the issue it had with old datasets

**Review by AdrianOlosutean — APPROVED** [review]
Looks good

**Review by HuvarVer — APPROVED** [review]
Build,
run,
tested MappingConformanceRule

**Pull request merged**